### PR TITLE
MULE-19424: Change default value for mule.http.requestStreaming.enable to true

### PR DESCRIPTION
**Pull request by eze210**

### CODE REVIEW

File: src/main/java/org/mule/service/http/impl/service/client/GrizzlyHttpClient.java [hunk 1]
```diff
@@ -272,7 +255,7 @@ protected void doConfigureProxy(AsyncHttpClientConfig.Builder builder, ProxyConf
     builder.setProxyServer(buildProxy(proxyConfig));
   }
 
-  protected final ProxyServer buildProxy(ProxyConfig proxyConfig) {
+  ProxyServer buildProxy(ProxyConfig proxyConfig) {
```
Comment (jcgrethe) [inline src/main/java/org/mule/service/http/impl/service/client/GrizzlyHttpClient.java:258]: public?

File: src/main/java/org/mule/service/http/impl/service/client/GrizzlyHttpClient.java [hunk 2]
```diff
@@ -621,7 +540,7 @@ protected void populateHeaders(HttpRequest request, RequestBuilder builder) {
     }
   }
 
-  private String getHostName() throws UnknownHostException {
+  String getHostName() throws UnknownHostException {
```
Comment (jcgrethe) [inline src/main/java/org/mule/service/http/impl/service/client/GrizzlyHttpClient.java:543]: public?

File: src/main/java/org/mule/service/http/impl/service/client/GrizzlyHttpClient.java [hunk 3]
```diff
@@ -656,43 +575,7 @@ private int retrieveMaximumHeaderSectionSize() {
...
-    }
-
-    public InputStreamBodyGenerator getInputStreamBodyGenerator() {
-      if (inputStream instanceof CursorStream) {
-        return new CursorInputStreamBodyGenerator((CursorStream) inputStream);
-      }
-
-      return new InputStreamBodyGenerator(inputStream);
-    }
+  private static boolean isRequestStreamingPropertySet() {
```
Comment (jcgrethe) [inline src/main/java/org/mule/service/http/impl/service/client/GrizzlyHttpClient.java:578]: why a method?

File: src/main/java/org/mule/service/http/impl/service/client/GrizzlyRequestConfigurer.java [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    if (request.getEntity() != null) {
+      if (request.getEntity().isStreaming()) {
+        setStreamingBodyToRequestBuilder(request, builder);
+      } else if (request.getEntity().isComposed()) {
+        for (HttpPart part : request.getEntity().getParts()) {
+          if (part.getFileName() != null) {
+            builder.addBodyPart(new ByteArrayPart(part.getName(), IOUtils.toByteArray(part.getInputStream()),
+                                                  part.getContentType(), null, part.getFileName()));
+          } else {
+            byte[] content = IOUtils.toByteArray(part.getInputStream());
```
Comment (jcgrethe) [inline src/main/java/org/mule/service/http/impl/service/client/GrizzlyRequestConfigurer.java:95]: static

File: src/main/java/org/mule/service/http/impl/service/client/GrizzlyRequestConfigurer.java [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+    builder.setFollowRedirects(!enableMuleRedirect && options.isFollowsRedirect());
+
+    client.populateHeaders(request, builder);
+
+    for (Map.Entry<String, String> entry : request.getQueryParams().entryList()) {
+      builder.addQueryParam(entry.getKey() != null ? encodeQueryElement(entry.getKey()) : null,
+                            entry.getValue() != null ? encodeQueryElement(entry.getValue()) : null);
+    }
+    options.getAuthentication().ifPresent((CheckedConsumer<HttpAuthentication>) (authentication -> {
+      Realm.RealmBuilder realmBuilder = new Realm.RealmBuilder()
```
Comment (jcgrethe) [inline src/main/java/org/mule/service/http/impl/service/client/GrizzlyRequestConfigurer.java:62]: static

**Review by jcgrethe — CHANGES_REQUESTED** [review]

File: src/test/java/org/mule/service/http/impl/service/client/GrizzlyRequestConfigurerTestCase.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.mule.runtime.http.api.client.HttpRequestOptions;
+import org.mule.runtime.http.api.domain.entity.HttpEntity;
+import org.mule.runtime.http.api.domain.message.request.HttpRequest;
+import org.mule.service.http.impl.service.client.GrizzlyHttpClient.RequestConfigurer;
+
+import java.io.IOException;
+
+@RunWith(MockitoJUnitRunner.class)
```
Comment (elrodro83) [inline src/test/java/org/mule/service/http/impl/service/client/GrizzlyRequestConfigurerTestCase.java:31]: use the tule instead of the runner

**Review by elrodro83 — CHANGES_REQUESTED** [review]

**Review by elrodro83 — APPROVED** [review]

**Review by jcgrethe — APPROVED** [review]

**Pull request merged**